### DISCUSSION TITLE
Issue with order_by using aggregate but the aggregate has args

### DISCUSSION THREAD

**Discussion by jiroRi** [hasura/graphql-engine#9020] [Question] [answered]
Hey there folks, I have an issue with using `order_by` in a query that has an aggregate like so:
```
query getPersonSuccessfulPurchase {
 persons(order_by: { orders_aggregate: { count: "desc_nulls_last" } }) {
  name
  orders_aggregate(where: { purchased_at: { _is_null: false } }) {
   aggregate {
    count
   }
  }
 }
}
```

The query in question results into something like: 

> John Smith
> pending purchases: 49
> successful purchases: 1
> total: 50

> John Doe
> pending purchases: 37
> successful purchases: 3
> total: 40

The key data here to look at is for both person's `successful purchases` and the `total` number of orders. I'm expecting that the data will be sorted by `successful purchases` but it is instead being sorted by `total` number of orders. Therefore resulting into John Smith being ahead of John Doe.

Any workaround on this?

**Comment by adas98012** [comment, marked as answer]
Can you provide more information about the data schema ? Especially on the "order" table ?

**Reply by jiroRi** [reply]
Hello @adas98012, thank you so much for checking

Let me know if this is what you need.. made a sample:
![image](https://user-images.githubusercontent.com/5771285/192987228-43f1fa99-b26e-4ac7-bb21-7be0ca623109.png)

More context: 
In this example, John Doe haven't successfully purchased anything yet(all *nulls*) while David Smith had one successful purchase record. If I do something like `order_by: { orders_aggregate: { count: "desc_nulls_last" } }`
while only counting successful purchases under orders `orders_aggregate(where: { purchased_at: { _is_null: false } })`

then I'm expecting that David will be ahead of John in the sortation since David had more successful purchases than John.. but, what happens is, the args in the orders_aggregate gets ignored.. instead, it counts "all the orders" regardless of whatever you put into the args

it then results into John Doe being ahead of David Smith in the sortation.

**Reply by adas98012** [reply]
I tried to reproduce the issue in my test system (Hasura Cloud: 2.12.0) , I got the expected result though (attaching screenshot).
The only difference in query is the "order_by" syntax (no quotes, shown with a pointer in the screenshot).


![Screen Shot 2022-09-29 at 1 41 44 PM](https://user-images.githubusercontent.com/101733366/193138596-ce6e48ac-7814-4b5f-9eb7-b9a6ed4e4ef8.png)

**Reply by jiroRi** [reply]
Could it be because of the hasura version? Been doing the query on Hasura `v1.3.3` :/

**Reply by jiroRi** [reply]
btw thank you so much for confirming on your end mate!
